src/net.rs

```diff
@@ -307,16 +326,22 @@ impl NetworkStream for HttpStream {
 
         match *self {
             HttpStream::Http(ref mut inner) => shutdown(&mut inner.0, how),
+            #[cfg(feature = "openssl")]
             HttpStream::Https(ref mut inner) => shutdown(&mut inner.get_mut().0, how)
         }
 
     }
 }
 
+#[cfg(feature = "openssl")]
 /// A connector that will produce HttpStreams.
 pub struct HttpConnector(pub Option<ContextVerifier>);
+#[cfg(not(feature = "openssl"))]
+/// A connector that will produce HttpStreams.
+pub struct HttpConnector(pub Option<()>);
```

Comment:
Any reason to keep the `Option<()>`? Without the verifier, it used to be a `struct HttpConnector;`.